lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -1284,6 +1294,38 @@ prop_MultiPoolCertificateSequence_coverage mpcs = checkCoverage
     certificateSequences = getSinglePoolSequences mpcs
     poolCount = length certificateSequences
 
+
+-- | read . put == pure
+prop_putHeaderListHeader
+    :: DBLayer IO
+    -> [BlockHeader]
+    -> Positive Int
+    -> Property
+prop_putHeaderListHeader DBLayer{..} headers (Positive k) =
+    monadicIO (setup >> prop)
+  where
+    sortedHeaders = sortBy (\a b -> compare (blockHeight a) (blockHeight b))
+        . L.nubBy (\a b -> (blockHeight a) == (blockHeight b)) . nub $ headers
+    expected = reverse . (if k == 0 then id else take k) . reverse $ sortedHeaders
+    setup = run $ atomically cleanDB
+    prop = do
+        void $ run $ atomically $ forM sortedHeaders putHeader
+        lHeaders <- run $ atomically $ listHeaders k
+        monitor $ counterexample $ unlines
+            [ "Read from DB: " <> show lHeaders
+            , "Expected: " <> show expected
+            , "Input: " <> show sortedHeaders
+            , "k: " <> show k]
+        monitor $ classify (null lHeaders) "Empty distributions"
+        assert (lHeaders == expected)
+        assert (strictlyIncreasing . fmap blockHeight $ lHeaders)
```

Comment:
I guess this property is kinda implicit in the fact that `expected` is already sorted that way, but I think it's fine to have it explicitly.